### ISSUE TITLE
Spurious error warning re `pkgbuild`

### ISSUE THREAD

**Issue by pearsonca** [richfitz/odin#318]
In using odin for the [paramix](https://github.com/cmmid/paramix) [demo analysis](https://github.com/cmmid/paramix/tree/main/inst/analysis), I'm seeing a non-breaking error message:

```
Loading required package: odin
Loading required namespace: pkgbuild
Failed with error:  ‘there is no package called ‘pkgbuild’’
```

This is from running the [simulate step](https://github.com/cmmid/paramix/blob/main/inst/analysis/scripts/simulate.R), which has a [separate script](https://github.com/cmmid/paramix/blob/main/inst/analysis/scripts/odin.R) to load / build the model.

Unclear how we should be avoiding seeing this spurious (I think?) error, or if there is some library bug which is erroneously raising this for our use case.

**Comment by richfitz** [comment]
I only see one place we non-fatally load pkgload and that's in `can_compile` so this may be coming from pkgload - a reprex might make that clearer.

I should warn you - we're about to do an extremely disruptive change that you might want to get ahead of (that may fix that issue).  We'll be swapping in the new version of odin soon (probably early Q1 2025) which has minor breaking changes to model code but significant changes to how models are run. See https://mrc-ide.github.io/odin2/articles/migrating.html and https://mrc-ide.github.io/dust2/articles/migrating.html for details.

**Comment by pearsonca** [comment]
Yeah, I've been trying to cook up a minimal example, but I also noticed after submitting this the impending odin2 overhaul. Suspect that'll either totally disappear this problem or otherwise reshape it into a different manifestation, so ... balancing my priorities.

To be somewhat cheeky, that package analysis should just work as a reprex (...after much renv gnashing of teeth), but I realize that isn't all that helpful.
